core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
Don't quite understand what you were recommending either. I thought you were saying:

a) Users writing Spark code will continue to use SparkContext way (apparently because that is the way they are used to). This will apply for jobs, smart connector.

b) Other cases (namely Snappy JDBC/ODBC) will use the new property.

(this doesn't have the Spark JDBC/ODBC case). But you are saying something else. Whats that and reasoning for the same?

My own preference is:

a) When using SnappySession always recommend using the property. Whether using smart connector or jobs (or Spark JDBC/ODBC in future when it will be enhanced to use SnappySession). And automatically applies to Snappy JDBC/ODBC.

b) When not using SnappySession, they will continue using the existing Spark way in any case.